### PR TITLE
Remove rootless functionality

### PR DESCRIPTION
**Pull request by winkingturtle-vmw**
- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
Remove abandoned rootless functionality. We deployed this to a cf-deployment and it was broken. Additionally fix path to secrets for running tests.

Related PRs:
- https://github.com/cloudfoundry/grootfs/pull/264
- https://github.com/cloudfoundry/guardian/pull/440
- https://github.com/cloudfoundry/dontpanic/pull/84
- https://github.com/cloudfoundry/garden-integration-tests/pull/137

Context:
- cloudfoundry/garden-runc-release#196
- cloudfoundry/garden-runc-release#316

Backward Compatibility
---------------
Breaking Change? **Yes**

Removal of rootless functionality that didn't work as expected.

### CODE REVIEW

File: scripts/create-docker-container.bash [hunk 1]
```diff
@@ -29,13 +29,27 @@ pushd "$CI/garden-runc-release/dockerfiles"
...
+  . "${HOME}/workspace/devenv/functions/gcp-secret-manager-helpers.bash"
+  export DOCKER_REGISTRY_USERNAME="$(gimme-secret-value-only dockerhub-tasruntime-username)"
+  export DOCKER_REGISTRY_PASSWORD="$(gimme-secret-value-only dockerhub-tasruntime-password)"
+fi
+if [[ "${DOCKER_REGISTRY_USERNAME:-undefined}" == "undefined" || "${DOCKER_REGISTRY_PASSWORD:-undefined}" == "undefined" ]]; then
+  cat << EOF
+  Run this script with DOCKER_REGISTRY_USERNAME, DOCKER_REGISTRY_PASSWORD env variables
+EOF
+exit 1
+fi
```
Comment (MarcPaquette) [inline scripts/create-docker-container.bash:42]: Should this functionality be included in an OSS CF repo?

File: scripts/fly/test.bash [hunk 2]
```diff
@@ -28,8 +28,8 @@ GARDEN_FUSE_TEST_ROOTFS=$PWD/input-01/garden-fuse.tar'
 
 if [[ -f "${HOME}/workspace/devenv/functions/gcp-secret-manager-helpers.bash" ]]; then
   . "${HOME}/workspace/devenv/functions/gcp-secret-manager-helpers.bash"
-  export DOCKER_REGISTRY_USERNAME="$(gimme-secret-value-only dockerhub-tasruntime | yq -r .user)"
-  export DOCKER_REGISTRY_PASSWORD="$(gimme-secret-value-only dockerhub-tasruntime | yq -r .password)"
+  export DOCKER_REGISTRY_USERNAME="$(gimme-secret-value-only dockerhub-tasruntime-username)"
+  export DOCKER_REGISTRY_PASSWORD="$(gimme-secret-value-only dockerhub-tasruntime-password)"
```
Comment (MarcPaquette) [inline scripts/fly/test.bash:32]: Another scenario where there is proprietary scripting.

File: src/greenskeeper/integration/greenskeeper_integration_test.go [hunk 3]
```diff
@@ -35,13 +35,10 @@ var _ = Describe("Greenskeeper", func() {
 				"PIDFILE=" + pidFileName,
 				"RUN_DIR=" + path.Join(tmpDir, "run"),
 				"GARDEN_DATA_DIR=" + path.Join(tmpDir, "garden"),
-				"CONTAINERD_DATA_DIR=" + path.Join(tmpDir, "containerd"),
```
Comment (MarcPaquette) [inline src/greenskeeper/integration/greenskeeper_integration_test.go:38]: Should there be a test for the presence of this env var in the other non-rootless tests?

File: scripts/create-docker-container.bash (hunk 1 above)
Comment (winkingturtle-vmw) [reply scripts/create-docker-container.bash:42]: I don't see an issue here. docker's username and password are required to run these tests. The option exists to set these env variables, but for those who have access to the cred, it's pulled from credential service.

File: src/greenskeeper/integration/greenskeeper_integration_test.go (hunk 3 above)
Comment (winkingturtle-vmw) [reply src/greenskeeper/integration/greenskeeper_integration_test.go:38]: hmm, I am not sure I follow your question. Can you rephrase your question?

Comment (MarcPaquette) [reply src/greenskeeper/integration/greenskeeper_integration_test.go:38]: https://github.com/cloudfoundry/garden-runc-release/blob/8002bd56ada80e24d9c81a22ba4bcbe629ce65c3/src/greenskeeper/integration/greenskeeper_integration_test.go#L70-L79

In "rootful" directories, we look for the other environment variables, but not for the `CONTAINERD_DATA_DIR` which seems odd. I wouldn't expect `CONTAINERD_DATA_DIR` to be contingent on rootless, and I think it would be a general env check.

Comment (MarcPaquette) [reply src/greenskeeper/integration/greenskeeper_integration_test.go:38]: In rootless we use containerd's datastore. In rootfull we use runc's datastore. Therefore we don't need to test for the existence of containerd env's in rootful.

**Review by MarcPaquette — APPROVED** [review]
